File: haskell/functional-ddd/src/Domain/OrderTaking.hs
{-# LANGUAGE EmptyDataDecls, KindSignatures #-}
module Domain.OrderTaking where


data CustomerId = CustomerId Int deriving (Show, Eq)
data WidgetCode = WidgetCode String deriving (Show, Eq)
data UnitQuantity = UnitQuantity Int deriving (Show, Eq)
data KilogramQuantity =  KilogramQuantity Float deriving (Show, Eq)

data Order = Order {
  customerInfo    :: CustomerInfo,
  shippingAddress :: ShippingAddress,
  billingAddress  :: BillingAddress,
  orderLines      :: [OrderLines],
  amountToBill    :: AmountToBill
} deriving (Show)

data CustomerInfo :: * -> *
data ShippingAddress :: * -> *
data BillingAddress :: * -> *
data OrderLines :: * -> *
data AmountToBill :: * -> *

File: kotlin/src/main/kotlin/domain/ordertaking/OrderTaking.kt
package domain.ordertaking

class CustomerId(val customerId: Int)
class WidgetCode(val widgetCode: String)
class UnitQuantity(val unitQuantity: Int)
class KilogramQuantity(val kilogramQuantity: Float)


data class Order(
    val customerInfo : Nothing,
    val shippingAddress : Nothing,
    val billingAddress : Nothing,
    val orderLines : Nothing,
    val amountToBill : Nothing
)